### DISCUSSION TITLE
How to add a grid submissioon command to the PyTest run?

### DISCUSSION THREAD

**Discussion by yurivict** [pytest-dev/pytest#12399] [General]
I would like to parallelize PyTest jobs beyond a single machine, and to run jobs on the grid.

Is there an easy way to add a grid submission command (ex. ```bsub -I -app main```) so that PyTest would prepend the command for each test with this parallelization command so that they would run remotely?

**Comment by RonnyPfannschmidt** [comment]
The ask is unclear, the grid tool you mentioned is unknown and pytest itself has xdist for multi machine testing 

There's also tools like pytest-split

**Comment by yurivict** [comment]
pytest-xdist distributes on multiple CPUs, AFAIK.

> [...] the grid tool you mentioned is unknown [...]

It doesn't have to be known. It's just some command that send another command to run on another machine in the same folder with the same environment. This is all information that is needed about this tool.
It can be called bsub, qsub, etc. qsub is from SGE - Sun Grid Environment.

All I need is a way to prepend the command that runs the individual test.

**Reply by RonnyPfannschmidt** [reply]
Theres nothing built-in to do what you ask and no tool to aggregate the results either 

As you are adamant in your belief that the provided information is sufficient, all I'm able to recommend is to use something like collect only to print out a list of test IDs,then externally submitting the pytest runs to the cluster

As that way of submission and gathering completely break appart reporting and stats as usually done,I would advise against doing that in general

Unless you have a reasonable way to manage the results aggregation,this will most likely be painful

**Comment by yurivict** [comment]
Wat do you mean by results aggregation?

The exit code and the printout of the command is the same.

The command is sent to the remote host, otherwise almost all effects of the command remain the same.
